src/IRCSock.cpp

```diff
@@ -1391,10 +1395,16 @@ void CIRCSock::ParseISupport(const CMessage& Message) {
             if (m_bNamesx) continue;
             m_bNamesx = true;
             PutIRC("PROTOCTL NAMESX");
+        } else if (sName.Equals("-NAMESX")) {
+            if (!m_bNamesx) continue;
```

Comment:
`continue` here is useless. In the true branch it prevented from sending `PROTOCTL` again, but here you don't send anything like `PROTOCTL -NAMESX`